include/nbl/builtin/hlsl/numeric_limits.hlsl

```diff
@@ -0,0 +1,202 @@
+    static const bool has_infinity = !is_integer;
+    
+    // (TODO) think about what this means for HLSL
+    // identifies floating-point types that can represent the special value "quiet not-a-number" (NaN)
+    static const bool has_quiet_NaN = false; 
+    // 	identifies floating-point types that can represent the special value "signaling not-a-number" (NaN)
+    static const bool has_signaling_NaN = false;
+    // 	identifies the denormalization style used by the floating-point type
+    static const float_denorm_style has_denorm = is_integer ? float_denorm_style::denorm_absent : float_denorm_style::denorm_present;
+    // identifies the floating-point types that detect loss of precision as denormalization loss rather than inexact result
+    static const bool has_denorm_loss = !is_integer;
+    // identifies the rounding style used by the type
+    static const float_round_style round_style = is_integer ? float_round_style::round_toward_zero : float_round_style::round_to_nearest;
+    // identifies the IEC 559/IEEE 754 floating-point types
+    static const bool is_iec559 = !is_integer;
+    // identifies types that represent a finite set of values
+    static const bool is_bounded = true;
+
+    // TODO verify this
+    // identifies types that handle overflows with modulo arithmetic
+    static const bool is_modulo = is_integer && !is_signed;
+
+    // number of radix digits that can be represented without change
+    static const int digits = !is_integer ? float_digits : (size - is_signed);
+    // number of decimal digits that can be represented without change
+    static const int digits10 = !is_integer ? float_decimal_digits : (size * 2 + size / 4 + int(!is_signed) * (size / 4););
+
+    // number of decimal digits necessary to differentiate all values of this type
+    static const int max_digits10 = !is_integer ? float_decimal_max_digits : 0;
+
+    // the radix or integer base used by the representation of the given type
+    static const int radix = 2;
+
+    // one more than the smallest negative power of the radix that is a valid normalized floating-point value
+    static const int min_exponent = !is_integer ? float_min_exponent : 0;
+    // the smallest negative power of ten that is a valid normalized floating-point value
+    static const int min_exponent10 = !is_integer ? float_min_decimal_exponent : 0;
+    // 	one more than the largest integer power of the radix that is a valid finite floating-point value
+    static const int max_exponent = !is_integer ? float_max_exponent : 0;
+    // the largest integer power of 10 that is a valid finite floating-point value
+    static const int max_exponent10 = !is_integer ? float_max_decimal_exponent : 0;
+
+    // identifies types which can cause arithmetic operations to trap
+    static const bool traps = false;
+    // identifies floating-point types that detect tinyness before rounding
+    static const bool tinyness_before = false;
```

Comment:
I'd rather move this into `numeric_limits` instead of keeping it in `num_base` cause from what I see `num_base` just provides a bunch of hacks to compile all this stuff efficiently

Also moving it out will allow you to break the inheritance so `numric_limits` doesnt `: num_base<T>` but `using _impl_base = impl::num_base<T>` internally instead which is cleaner IMHO